opentech/apply/users/urls.py

```diff
@@ -11,9 +13,8 @@
 public_urlpatterns = [
     path(
         'login/',
-        auth_views.LoginView.as_view(
-            template_name='users/login.html',
-            redirect_authenticated_user=True
```

Comment:
The two_factor LoginView does not support "redirect_authenticated_user" setting. Any idea how to get the functionality back?

Reply:
We need to extend the Django 2fa [LoginView](https://github.com/Bouke/django-two-factor-auth/blob/87b644ce5bbd9f65959bd619b1782868f14888d3/two_factor/views/core.py#L51) for [redirecting](https://github.com/django/django/blob/b9cf764be62e77b4777b3a75ec256f6209a57671/django/contrib/auth/views.py#L53) authenticated users to Login Redirect URL say dashboard. 

Or we can also change template code so `My OTF` button detects the user is logged-in and take user to dashboard instead of the login page.

Reply:
Implemented this now but it feels like I have duplicated a lot of code, better way do do it?

Reply:
Looks good, some code duplication is required. We also need to add other method decorators to dispatch method as they are not carried over from parent 2fa login view.

```python
    @method_decorator(sensitive_post_parameters())
    @method_decorator(csrf_protect)
    @method_decorator(never_cache)
    def dispatch(self, request, *args, **kwargs):
    ....
    ....
```

Reply:
Fixed now, thanks for the ping about it.